IPython/core/logger.py

```diff
@@ -15,10 +15,15 @@
 # Python standard modules
 import glob
 import io
+import logging
 import os
 import time
 
 
+
+# prevent jedi/parso's debug messages pipe into interactiveshell
+logging.getLogger("parso").setLevel(logging.WARNING)
```

Comment:
I'm not a maintainer here, just a lurker. But I think imports with side-effects are bad. Better put this in a function and call it from some point in IPython initialization.

Reply:
@intgr Would you suggest a place to call the function with line number in interactiveshell.py?

Look at history.py & aliases.py, there are variables assignments in the module level, too.
I think this is acceptable in ipython.
And this change is not that important to change the flow of interactiveshell.py, so IMO this way is more elegant and won't mess with interactiveshell.py.

Reply:
I don't know, sorry.

Reply:
Thanks for the review and the discussion, in general we don't like side effect, but as IPython is a shell/interpreter often we don't have too much choices to give a good user experience. I think that is fine here for the time being unless we find a better place.